### PR TITLE
`IsNotEmpty` operation

### PR DESCRIPTION
**Pull request by ktrzos**
This PR:

- [x] Provide a new `IsNotEmpty` operation 
- [x] Is covered by unit tests
- [x] Has static analysis tests (psalm, phpstan)
- [x] Has documentation

Related to #291.

### CODE REVIEW

File: src/Operation/IsNotEmpty.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @template T
+ */
+final class IsNotEmpty extends AbstractOperation
+{
+    /**
+     * @return Closure(iterable<TKey, T>): Generator<int, bool>
+     */
+    public function __invoke(): Closure
+    {
+        return static fn (iterable $iterable): Generator => yield (bool)(new IterableIteratorAggregate($iterable))->getIterator()->valid();
```
Comment (drupol) [inline src/Operation/IsNotEmpty.php:24]: Is the `(bool)` cast mandatory here ?
Is `->valid()` not returning a boolean already?

Comment (ktrzos) [reply src/Operation/IsNotEmpty.php]: Yes, you're right. Changed.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi,

Thanks for the PR, everything seems ok, I don't have much to say.

However, there's one missing thing.

Since we include a `CollectionDecorator`, you need to add the new method in it as well.

Once that issue and the other minor inline comment I did is done, I'll wait 24h before merging it, so @AlexandruGG is able to review it as well.

Once again, thanks for taking the time to do that ! 

PS: Do not mind the static analysis issues with PSalm, we are working on it in #287

**Comment by drupol** [comment]
@ktrzos Are you at ease with the requested changes? Feel free to ask more info if needed!

**Comment by ktrzos** [comment]
@drupol 

Sorry. Didn't have much time during the last days. But things are fixed now 🙂.

**Comment by drupol** [comment]
It's super fine, no worries.

Thanks for fixing it !

File: src/Contract/Operation/IsNotEmptyable.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace loophp\collection\Contract\Operation;
+
+/**
+ * @template TKey
+ * @template T
+ */
+interface IsNotEmptyable
+{
+    /**
+     * Check if a collection has any elements inside.
```
Comment (AlexandruGG) [inline src/Contract/Operation/IsNotEmptyable.php:14]: please make this doc block description the exact same as the one in the docs [here](https://github.com/loophp/collection/pull/292/files?diff=split&w=1#diff-89f6324da13155a697ae0b42136e03a40892dd4edc4257ea1e451f0ea2cc3f80R1283)

File: docs/pages/api.rst [hunk 3]
```diff
@@ -1268,7 +1268,7 @@ Signature: ``Collection::intersperse($element, int $every = 1, int $startAt = 0)
 isEmpty
 ~~~~~~~
 
-Check if a collection has any elements inside.
+Check if a collection has no elements inside.
```
Comment (AlexandruGG) [inline docs/pages/api.rst:1271]: I like this distinction, but please update the one on the `IsEmptyable` interface as well :)

**Review by AlexandruGG — APPROVED** [review]
Thanks for the contribution! 👍 

A couple of minor comments - I'll approve but please update the docs accordingly.

**Comment by ktrzos** [comment]
@AlexandruGG 
Good catch! Updated 👍️.

**Comment by drupol** [comment]
Thanks !!!

I hope it was not too complex! Feel free to come up with new additions!

**Pull request merged**